### ISSUE TITLE
List bucket fails with error

### ISSUE THREAD

**Issue by spradeep** [rustyconover/net-amazon-s3#77]
```
my $s3 = Net::Amazon::S3->new(
    {   aws_access_key_id     => $aws_access_key_id,
        aws_secret_access_key => $aws_secret_access_key,
        retry                 => 1,
    }
);
 
# a bucket is a globally-unique directory
# list all buckets that i own
my $response = $s3->buckets;
foreach my $bucket ( @{ $response->{buckets} } ) {
    print "You have a bucket: " . $bucket->bucket . "\n";
}
```
Returns error Can't call method "region" on an undefined value at /usr/local/share/perl5/Net/Amazon/S3/HTTPRequest.pm line 49.

Version : 0.89

**Comment by happy-barney** [comment]
Thanks for reporting,
Regression introduced with region request optimization.

**Comment by happy-barney** [comment]
Fixed by #78
